plugins/outputs/iotdb/iotdb.go

```diff
@@ -0,0 +1,277 @@
+	}
+	s.Log.Info("IoTDB output plugin initialization completed.")
+	return nil
+}
+
+func (s *IoTDB) Connect() error {
+	sessionConf := &client.Config{
+		Host:     s.Host,
+		Port:     s.Port,
+		UserName: s.User,
+		Password: s.Password,
+	}
+	var ss = client.NewSession(sessionConf)
+	s.session = &ss
+	if err := s.session.Open(false, int(time.Duration(s.Timeout)/time.Millisecond)); err != nil {
+		return fmt.Errorf("connecting to %s:%s failed: %w", s.Host, s.Port, err)
+	}
+	return nil
+}
+
+func (s *IoTDB) Close() error {
+	// Close any connections here.
+	// Write will not be called once Close is called, so there is no need to synchronize.
+	_, err := s.session.Close()
+	return err
+}
+
+// Write should write immediately to the output, and not buffer writes
+// (Telegraf manages the buffer for you). Returning an error will fail this
+// batch of writes and the entire batch will be retried automatically.
+func (s *IoTDB) Write(metrics []telegraf.Metric) error {
+	// Convert Metrics to Records with Tags
+	rwt, err := s.ConvertMetricsToRecordsWithTags(metrics)
+	if err != nil {
+		return err
+	}
+	// Write to client.
+	// If first writing fails, the client will automatically retry three times. If all fail, it returns an error.
+	if err := s.WriteRecordsWithTags(rwt); err != nil {
+		s.Log.Errorf("IoTDB Write Error: %s", err.Error())
+		return err
+	}
+	return nil
+}
+
+// Find out data type of the value and return it's id in TSDataType, and convert it if necessary.
+func (s *IoTDB) getDataTypeAndValue(value interface{}) (client.TSDataType, interface{}) {
+	switch v := value.(type) {
+	case int32:
+		return client.INT32, v
+	case int64:
+		return client.INT64, v
+	case uint32:
+		return client.INT64, int64(v)
+	case uint64:
+		switch s.ConvertUint64To {
+		case "int64_clip":
+			if v <= uint64(math.MaxInt64) {
+				return client.INT64, int64(v)
+			}
+			return client.INT64, int64(math.MaxInt64)
+		case "int64":
+			return client.INT64, int64(v)
+		case "text":
+			return client.TEXT, fmt.Sprintf("%d", v)
+		default:
+			return client.UNKNOW, int64(0)
+		}
+	case float64:
+		return client.DOUBLE, v
+	case string:
+		return client.TEXT, v
+	case bool:
+		return client.BOOLEAN, v
+	default:
+		s.Log.Errorf("Unknown datatype: '%T' %v", value, value)
+		return client.UNKNOW, int64(0)
+	}
+}
+
+// convert Timestamp Unit according to config
+func (s *IoTDB) convertTimestampOfMetric(m telegraf.Metric) (int64, error) {
+	switch s.TimeStampUnit {
+	case "second":
+		return m.Time().Unix(), nil
+	case "millisecond":
+		return m.Time().UnixMilli(), nil
+	case "microsecond":
+		return m.Time().UnixMicro(), nil
+	case "nanosecond":
+		return m.Time().UnixNano(), nil
+	default:
+		return 0, fmt.Errorf("IoTDB Configuration Error: unknown TimeStampUnit: %s", s.TimeStampUnit)
+	}
+}
+
+// convert Metrics to Records with tags
+func (s *IoTDB) ConvertMetricsToRecordsWithTags(metrics []telegraf.Metric) (*recordsWithTags, error) {
```

Comment:
Do not export this function
```suggestion
func (s *IoTDB) convertMetricsToRecordsWithTags(metrics []telegraf.Metric) (*recordsWithTags, error) {
```

Reply:
It's fixed.